### ISSUE TITLE
Allow passing optional service manager to `createApp()`

### ISSUE THREAD

**Issue by jeredmasters** [FoalTS/foal#598]
### Purpose
Provides the ability to configure different service classes to be loaded at application start.

### Use case
Deploying a sandbox version of the app is difficult where all functionality is essentially the same, however all requests to a specific thrid party service can (and should) be mocked. These are all contained in one service, which it would be incredibly powerful and elegant to be able to swap this out for the mock service before the rest fo the application loads.

In this case the real service class and the mock both implement the same TS interface, so consistiency is guaranteed.

### Suggestion
Augment the `createApp()` function to accept an optional parameter for `serviceManager`:
```typescript
export function createApp(
  rootControllerClass: Class,
  expressInstanceOrOptions?: ExpressApplication | ExpressOptions,
  serviceManager?: ServiceManager
){
  ...
}
```

### Example useage
```typescript
const serviceManager = new ServiceManager();
if (environment === 'SANDBOX')
  serviceManager.set({
    thirdPartyApi: new MockThirdyParty()
  });
}
const app = createApp(RootController, {}, serviceManager);
```

### Other notes
After implementing this locally is became tempting to replace the dependencies with the interface, instead of the primary class. Then use the above logic to set the mapped indentity for both cases. 
eg:
```typescript
interface IThirdPartyApi {
  mutateState(newState: State): ChangeVector;
}
class ThirdPartyApi implments IThirdPartyApi {
  mutateState(newState: State): ChangeVector { /* ... */ }
}
class MockThirdPartyApi implments IThirdPartyApi {
  mutateState(newState: State): ChangeVector {
    switch(newState){ /* set of known responses */ }
  }
}
export class SomeInternalBusinessLogic {
  @dependency
  thirdPartyApi: IThirdPartyApi;
}
```

This leads to the idea of replacing all dependencies with interface version, and manually constructing the whole `ServiceManager` identity mapper.

**Comment by LoicPoullain** [comment]
Reopening. Issue not solved completely: see #627

**Comment by LoicPoullain** [comment]
Feature added in v1.8.
